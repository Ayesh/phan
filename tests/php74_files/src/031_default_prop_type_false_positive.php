<?php
class test31 {
    public float $x = 0;
    public float $y = 0.0;
    public float $z = '0';
    public float $w = null;
    public ?float $w2 = null;
}

$y = new test31;
// Setting $x to an int will not raise any warnings
$y->x = 5;
var_dump($y->x);
