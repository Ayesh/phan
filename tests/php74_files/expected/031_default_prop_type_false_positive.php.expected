%s:5 PhanTypeMismatchPropertyDefaultReal Default value for float $z can't be '0' of type string
%s:6 PhanTypeMismatchPropertyDefaultReal Default value for float $w can't be null of type null
